mlx/backend/common/binary.h

```diff
@@ -43,13 +43,18 @@ void set_binary_op_output_data(
     array& out,
     BinaryOpType bopt,
     bool donate_with_move = false) {
+  constexpr size_t donation_extra = 16384;
+  bool b_donatable = b.is_donatable() && b.itemsize() == out.itemsize() &&
+      b.buffer_size() <= out.nbytes() + donation_extra;
+  bool a_donatable = a.is_donatable() && a.itemsize() == out.itemsize() &&
+      a.buffer_size() <= out.nbytes() + donation_extra;
```

Comment:
It might be worth encapsulating this logic somewhere so that we can re-use it in unary / binary_two/ ternary / everywhere else that we have potential donation. (I think it is also possible in those cases to donate overly large buffers?)

Reply:
Is something like that what you had in mind?

Reply:
Exactly 👍